internal/mesh/internal/controllers/sidecarproxy/builder/destination_builder.go

```diff
@@ -26,7 +26,12 @@ import (
 func (b *Builder) BuildDestinations(destinations []*intermediate.Destination) *Builder {
 	var lb *ListenerBuilder
 	if b.proxyCfg.IsTransparentProxy() {
-		lb = b.addTransparentProxyOutboundListener(b.proxyCfg.DynamicConfig.TransparentProxy.OutboundListenerPort)
+		if b.proxyCfg.DynamicConfig.TransparentProxy == nil {
+			// TODO: should this be defaulted elsewhere?
+			lb = b.addTransparentProxyOutboundListener(15001)
```

Comment:
Could you do it in `func (f *Fetcher) FetchAndMergeProxyConfigurations(ctx context.Context, id *pbresource.ID) (*pbmesh.ProxyConfiguration, error) {`?

Reply:
Nice, that seems good